### ISSUE TITLE
rust-wasm: Imported resources should implement `Clone`

### ISSUE THREAD

**Issue by Liamolucko** [bytecodealliance/wit-bindgen#136]
The `resource_clone_*` functions aren't currently exposed to Rust for imported resources. I think they should be exposed as a `Clone` implementation, like they already are on `Handle` for exported resources.

**Comment by alexcrichton** [comment]
Yeah one thing that isn't really handled well today is how imported/exported resources are treated differently. I think they should probably be using the same underlying abstractions but I haven't had a chance to review and update this. In the meantime though I definitely agree with adding `Clone`!
